### DISCUSSION TITLE
How to call openai's api in vanna when using vpn

### DISCUSSION THREAD

**Discussion by hhpyy** [vanna-ai/vanna#247] [Q&A]
Hello, I am deploying vanna locally, but I get a "Connection Error" error. Most likely, I am unable to call the openai api. I saw online that I can change the api_requestor.py file of the openai library, but I don't see this file in my files（My instructions are ‘ pip install 'vanna[chromadb,openai,postgres]'’）

How can I add an agent in Vanna? (Just like in the picture.)Or how can I connect to openai’s api?thank you for your reply
![2024-01-31](https://github.com/vanna-ai/vanna/assets/104671469/8231f941-6038-4e6d-878d-7f6076a25948)

**Comment by xiguazhiPrince** [comment]
hi，are you solve it?
